hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeysRenameRequest.java

```diff
@@ -0,0 +1,289 @@
+import org.apache.hadoop.ozone.security.acl.OzoneObj;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.hadoop.ozone.om.exceptions.OMException.ResultCodes.KEY_NOT_FOUND;
+
+/**
+ * Handles rename keys request.
+ */
+public class OMKeysRenameRequest extends OMKeyRequest {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OMKeysRenameRequest.class);
+
+  public OMKeysRenameRequest(OMRequest omRequest) {
+    super(omRequest);
+  }
+
+  /**
+   * Stores the result of request execution for Rename Requests.
+   */
+  private enum Result {
+    SUCCESS,
+    DELETE_FROM_KEY_ONLY,
+    REPLAY,
+    FAILURE,
+  }
+
+  @Override
+  public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
+
+    RenameKeysRequest renameKeysRequest = getOmRequest().getRenameKeysRequest();
+    Preconditions.checkNotNull(renameKeysRequest);
+
+    return getOmRequest().toBuilder()
+        .setRenameKeysRequest(renameKeysRequest.toBuilder())
+        .setUserInfo(getUserInfo()).build();
+
+  }
+
+  @Override
+  @SuppressWarnings("methodlength")
+  public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
+      long trxnLogIndex, OzoneManagerDoubleBufferHelper omDoubleBufferHelper) {
+
+    RenameKeysRequest renameKeysRequest = getOmRequest().getRenameKeysRequest();
+    OMClientResponse omClientResponse = null;
+    Set<OmKeyInfo> unRenamedKeys = new HashSet<>();
+    List<OmRenameKeyInfo> renameKeyInfoList = new ArrayList<>();
+
+    OMMetrics omMetrics = ozoneManager.getMetrics();
+    omMetrics.incNumKeyRenames();
+
+    AuditLogger auditLogger = ozoneManager.getAuditLogger();
+
+
+    OMResponse.Builder omResponse = OmResponseUtil.getOMResponseBuilder(
+        getOmRequest());
+
+    OMMetadataManager omMetadataManager = ozoneManager.getMetadataManager();
+    IOException exception = null;
+    OmKeyInfo fromKeyValue = null;
+
+    Result result = null;
+    Map<String, String> auditMap = null;
+    RenameKeyRequest renameRequest = null;
+    String toKey = null;
+    String fromKey = null;
+    String volumeName = null;
+    String bucketName = null;
+    String fromKeyName = null;
+    String toKeyName = null;
+    try {
+      for (RenameKeyRequest renameKeyRequest : renameKeysRequest
+          .getRenameKeyRequestList()) {
+        OzoneManagerProtocolProtos.KeyArgs renameKeyArgs =
+            renameKeyRequest.getKeyArgs();
+        volumeName = renameKeyArgs.getVolumeName();
+        bucketName = renameKeyArgs.getBucketName();
+        fromKeyName = renameKeyArgs.getKeyName();
+        String objectKey = omMetadataManager.getOzoneKey(volumeName, bucketName,
+            fromKeyName);
+        OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable().get(objectKey);
+        unRenamedKeys.add(omKeyInfo);
```

Comment:
unRenamedKeys is not updated inside the loop. Can we add a unit test for the returned unRenamedKeys upon failure?

Reply:
Although we have put the list of unDeletedKeys and unRenamedKeys into the response. Currently, renameKeys and DeleteKeys in OzoneBucket.java and ClientProtocol.java are still void. 
I've added TODO to the OMClientRequest and created a jira(HDDS-3916), and I'll change those separately.